### ISSUE TITLE
last_epoch parameter of CyclicLR and OneCycleLR is not the number of epochs

### ISSUE THREAD

**Issue by NicolasHug** [pytorch/pytorch#69753]
From the [docs](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CyclicLR.html#torch.optim.lr_scheduler.CyclicLR), `last_epoch` in `CyclicLR` and `OneCycleLR` corresponds to the number of batches (or the number of steps / updates), not the number of epoch:

> last_epoch (int) – The index of the last batch. This parameter is used when resuming a training job. Since step() should be invoked after each batch instead of after each epoch, this number represents the total number of batches computed, not the total number of epochs computed. When last_epoch=-1, the schedule is started from the beginning. Default: -1

A lot of the other scheduler also accept a `last_epoch` parameter, which is the actual number of epochs, as expected.

In the original PR https://github.com/pytorch/pytorch/pull/2016 which was superseded, this paramater was named `last_batch_idx`, which seems more descriptive.

Is there a specific reason why this ended up being `last_epoch`? If not, should we consider deprecate this parameter and rename it to `last_batch_idx` instead?
